crates/wadm/src/scaler/spreadscaler/link.rs

```diff
@@ -149,12 +149,15 @@ where
                 (
                     true,
                     // TODO(#88): reverse compare too
-                    // Ensure all named configs are the same
-                    linkdef.source_config.iter().all(|config_name| {
-                        self.config.source_config.iter().any(|c| c == config_name)
-                    }) || linkdef.target_config.iter().all(|config_name| {
-                        self.config.target_config.iter().any(|c| c == config_name)
-                    }),
+                    // Ensure all supplied configs are the same
+                    linkdef
+                        .source_config
+                        .iter()
+                        .eq(self.config.source_config.iter())
+                        || linkdef
```

Comment:
Is this a bug in the original implementation, should this be a `&&`?

e.g. all source config is the same & all target config is the same?

Reply:
Yeah actually I thought it was -- but wasn't sure... Is there any reason you'd want one *or* the other...?

Reply:
Maybe the original logic was something like `if source_different || target_different`